src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,180 @@
+        flex-direction: row;
+    }
+`;
+
+const StyledStar = styled(StarIcon)`
+    width: ${size.mediumLarge};
+    height: ${size.mediumLarge};
+    margin-right: ${size.mediumLarge};
+    box-sizing: content-box;
+
+    @media ${screenSize.mediumAndUp} {
+        padding: 4px;
+        width: auto;
+        height: auto;
+        margin-right: ${({ isActive }) => (isActive ? '0' : size.xsmall)};
+
+        &:hover {
+            cursor: ${({ isClicked }) => (isClicked ? 'default' : 'pointer')};
+        }
+    }
+`;
+
+const StyledStarsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    @media ${screenSize.mediumAndUp} {
+        margin-left: ${size.large};
+    }
+`;
+
+const StyledTitle = styled.span`
+    font-family: 'Fira Mono', monospace;
+    font-weight: bold;
+    font-size: ${fontSize.small};
+`;
+
+const initialState = {
+    first: false,
+    second: false,
+    third: false,
+    fourth: false,
+    fifth: false,
+    isClicked: false,
+};
+
+const reducer = (state, action) => {
+    let nextState = { ...initialState };
+    switch (action) {
+        case 'fifth':
+            nextState = { ...nextState, fifth: true };
+        case 'fourth':
+            nextState = { ...nextState, fourth: true };
+        case 'third':
+            nextState = { ...nextState, third: true };
+        case 'second':
+            nextState = { ...nextState, second: true };
+        case 'first':
+            nextState = { ...nextState, first: true };
+            break;
+        case 'clicked':
+            nextState = { ...state, isClicked: true };
+            break;
+        case 'clear':
+            nextState = { ...initialState };
+            break;
+        default:
+            throw new Error();
+    }
+
+    return nextState;
+};
+
+const StarRating = ({ onFirstsClick, onMiddleClick, onLatestClick }) => {
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const onClickHandler = useCallback(
+        handler => {
+            if (!state.isClicked) {
+                handler && handler();
+                dispatch('clicked');
+            }
+        },
+        [state.isClicked]
+    );
+
+    const onHoverHandler = useCallback(
+        star => {
+            !state.isClicked && dispatch(star);
+        },
+        [state.isClicked]
+    );
+    const onLeaveHandler = useCallback(() => {
+        !state.isClicked && dispatch('clear');
+    }, [state.isClicked]);
+
+    return (
+        <StyledContainer>
+            <StyledTitle>Rate this article</StyledTitle>
+            <StyledStarsContainer onMouseLeave={onLeaveHandler}>
+                <StyledStar
+                    isActive={state.first}
+                    isClicked={state.isClicked}
+                    name="first-star"
+                    onClick={() => onClickHandler(onFirstsClick)}
+                    onMouseEnter={() => onHoverHandler('first')}
+                    width={state.first && size.mediumLarge}
```

Comment:
I noticed when we increase the size it seems to push to the right. We would like the star to stay centered in its bounding box but expand from the center. If we hover over it we can see this happening:

![Screen Shot 2021-04-06 at 10 37 13 AM](https://user-images.githubusercontent.com/9064401/113728442-1c733200-96c4-11eb-996e-fd8a342f0714.png)
![Screen Shot 2021-04-06 at 10 37 20 AM](https://user-images.githubusercontent.com/9064401/113728443-1c733200-96c4-11eb-90f3-2e1acd23163d.png)